crates/storage/provider/src/traits/chain.rs

```diff
@@ -139,3 +139,44 @@ impl CanonStateNotification {
         receipts
     }
 }
+
+/// Wrapper around a broadcast receiver that receives fork choice notifications.
+#[derive(Debug)]
+pub struct ForkChoiceNotifications(broadcast::Receiver<SealedHeader>);
+
+/// A trait that allows to register to fork choice related events
+/// and get notified when a new fork choice is available.
+pub trait ForkChoiceSubscriptions: Send + Sync {
+    /// Get notified when a new head of the chain is selected.
+    fn subscribe_to_fork_choice(&self) -> ForkChoiceNotifications;
+
+    /// Convenience method to get a stream of the new head of the chain.
+    fn fork_choice_stream(&self) -> ForkChoiceStream {
+        ForkChoiceStream { st: BroadcastStream::new(self.subscribe_to_fork_choice().0) }
+    }
+}
+
+/// A stream of the fork choices in the form of [`SealedHeader`].
+#[derive(Debug)]
+#[pin_project::pin_project]
+pub struct ForkChoiceStream {
+    #[pin]
+    st: BroadcastStream<SealedHeader>,
+}
+
+impl Stream for ForkChoiceStream {
+    type Item = SealedHeader;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        loop {
+            return match ready!(self.as_mut().project().st.poll_next(cx)) {
+                Some(Ok(notification)) => Poll::Ready(Some(notification)),
+                Some(Err(err)) => {
+                    debug!(%err, "finalized header notification stream lagging behind");
+                    continue
+                }
```

Comment:
```suggestion
                Some(Err(err)) => {
                    debug!(target: "provider::fcu",
                        %err,
                        "finalized header notification stream lagging behind"
                    );
                    
                    cx.waker().wake_by_ref();
                    return Poll::Pending
                }
```
if we don't return the error here

Reply:
should I also modify `CanonStateNotificationStream`?

Reply:
pushed without for now, if needed I can make a follow up PR

Reply:
yeah, please modify `CanonStateNotificationStream` in a new PR, you can ref this comment in the PR description

Reply:
thanks!